src/main/java/org/redcrosswarriors/controller/RequestBloodController.java

```diff
@@ -0,0 +1,52 @@
+package org.redcrosswarriors.controller;
+import org.redcrosswarriors.controllerservice.RequestBloodControllerService;
+import org.redcrosswarriors.model.input.RequestBloodInput;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.security.Principal;
+import java.util.HashMap;
+import java.util.Map;
+
+@RestController
+public class RequestBloodController
+{
+    @Autowired
+    RequestBloodControllerService requestBloodControllerService;
+
+
+
+
+
+    @PostMapping("/bloodRequest")
+    @Secured("ROLE_USER")
+    public ResponseEntity<Object> requestBlood(@RequestBody RequestBloodInput input)
+    {
+        System.out.println("start");
+
+
+
+        Map<String, Object> json = new HashMap();
+        if(requestBloodControllerService.requestBlood(input))
+        {
+            System.out.println("start");
+            json.put("status", "Request Processed completely");
+            int numbers = requestBloodControllerService.numberOfMatches(input.getBloodType());
```

Comment:
You should return the number of matches in the json response. Right now it's not really doing anything but printing to the console.

Reply:
yes I am planning to do it after the merge, but I will make sure I will finish it tonight itself.